src/riscv-integration.adoc

```diff
@@ -228,17 +228,18 @@ instruction following the jump (<<pcc>> + 4) is written to a *c* register.
 The capability jump and link register (<<CJALR>>) instruction replaces the jump
 and link register (<<JALR>>) instruction at the same encoding. This instruction
 allows unconditional jumps to a target capability. The target capability is
-obtained by incrementing the capability in the *c* register operand by the
-sign-extended 12-bit immediate, then setting the least significant bit of the
-result to zero. The capability with the address of the instruction following
-the jump (<<pcc>> + 4) is written to a *c* register.
+obtained by unsealing and incrementing the capability in the *c* register
```

Comment:
Unsealing happens last, so that any address manipulation invalidates it

Reply:
Adding immediate 0 unseals according to the CINCOFFSETIMM page, so the other order doesn't work.

Reply:
Logically the add is conditional on the immediate being non-zero. Of course that's not how you'd implement it, but that's how it should be specified. Talking about "unsealing and incrementing" is a terrifying prospect as it's just asking for people to misinterpret it and provide a gadget for modifying sealed capabilities.

Reply:
You can think of it as similar to the rules around legalisation, where there's a conditional CSetAddr

Reply:
I agree with @jrtc27: Unsealing should happen last to avoid misinterpretation.

The idea of this paragraph was to describe how the address part of the capability is obtained. The sealing part of the instruction was left out, so that needs fixing. Note that the sealing of the return address is also missing.